db/migrate/20200910143249_fix_course_application_open_from_dates.rb

```diff
@@ -0,0 +1,14 @@
+class FixCourseApplicationOpenFromDates < ActiveRecord::Migration[6.0]
+  def up
+    recruitment_cycle = RecruitmentCycle.find_by(year: "2021")
+    return if recruitment_cycle.nil?
+
+    courses = recruitment_cycle.courses.where.not(applications_open_from: recruitment_cycle.application_start_date..recruitment_cycle.application_end_date)
+
+    courses.update_all(applications_open_from: recruitment_cycle.application_start_date)
+  end
+
+  def down
+    # cannot go back as this will be invalid
```

Comment:
We can make this reversible if we want to be more cautious. We could add a new column e.g. `old_applications_from_value` and copy the existing values to it before the update. Then drop the column in a subsequent migration in a future PR once we're happy this one didn't cause issues. Whether we do that depends on how happy we are with the risk. Might be overkill here 🤷

Reply:
Discussed offline. We're going to proceed as is with this and review carefully in staging.